src/Mvc/Mvc.Core/src/ProducesResponseTypeAttribute.cs

```diff
@@ -57,10 +87,37 @@ public ProducesResponseTypeAttribute(Type type, int statusCode)
         /// <value></value>
         internal bool IsResponseTypeSetByDefault { get; }
 
+        // Internal for testing
+        internal MediaTypeCollection ContentTypes => _contentTypes;
+
         /// <inheritdoc />
         void IApiResponseMetadataProvider.SetContentTypes(MediaTypeCollection contentTypes)
         {
-            // Users are supposed to use the 'Produces' attribute to set the content types that an action can support.
+            contentTypes.Clear();
+            foreach (var contentType in _contentTypes)
+            {
+                contentTypes.Add(contentType);
+            }
+        }
+
+        private static MediaTypeCollection GetContentTypes(string contentType, string[] additionalContentTypes)
+        {
+            List<string> completeContentTypes = new(additionalContentTypes.Length + 1);
```

Comment:
I think we still prefer using `var` over this pattern. Can we keep that until we decide this is cool?